pvlib/solarposition.py

```diff
@@ -1445,3 +1454,96 @@ def sun_rise_set_transit_geometric(times, latitude, longitude, declination,
     sunset = _local_times_from_hours_since_midnight(times, sunset_hour)
     transit = _local_times_from_hours_since_midnight(times, transit_hour)
     return sunrise, sunset, transit
+
+
+def spencer_mc(times, latitude, longitude):
+    """
+    Calculate the solar position using a python implementation of the
+    Spencer (1972) formulation provided by meteocontrol
+
+    Parameters
+    ----------
+    times : :class:`pandas.DatetimeIndex`
+        Corresponding timestamps, must be localized to the timezone for the
+        ``latitude`` and ``longitude``.
+    latitude : float
+        Latitude in degrees, positive north of equator, negative to south
+    longitude : float
+        Longitude in degrees, positive east of prime meridian, negative to west
+
+    Returns
+    -------
+    DataFrame
+        The DataFrame will have the following columns:
+        zenith (degrees),
+        elevation (degrees),
+        azimuth (degrees),
+        equation_of_time (seconds),
+        eccentricity,
+        declination (degrees).
+
+    References
+    ----------
+    [1] Spencer (1972) can be found in
+    Sen, Zekai. Solar energy fundamentals and modeling techniques:
+     atmosphere, environment, climate change and renewable energy.
+     Springer Science & Business Media, 2008.
+    """
+
+    julians = datetime_to_julian(times)
+    julians_2000 = np.asarray(julians, dtype=np.float) - JULIAN_2000
+
+    lat = np.radians(latitude)
+
+    # Compute fractional year (gamma) in radians
+    gamma = 2 * np.pi * (julians_2000 % JULIAN_YEARS) / JULIAN_YEARS
+    cos_gamma = np.cos(gamma), np.cos(gamma * 2), np.cos(gamma * 3)
+    sin_gamma = np.sin(gamma), np.sin(gamma * 2), np.sin(gamma * 3)
+    day_time = (julians_2000 % 1) * 24
+
+    # Eccentricity: correction factor of the earth's orbit.
+    eccentricity = 1.00011 + 0.034221 * cos_gamma[0]
+    eccentricity += 0.001280 * sin_gamma[0]
+    eccentricity += 0.000719 * cos_gamma[1]
+    eccentricity += 0.000077 * sin_gamma[1]
+
+    declination = np.array(declination_spencer71(times.dayofyear))
+
+    # Equation of time (difference between standard time and solar time).
+    eot = 0.000075 + 0.001868 * cos_gamma[0]
+    eot -= 0.032077 * sin_gamma[0]
+    eot -= 0.014615 * cos_gamma[1]
+    eot -= 0.040849 * sin_gamma[1]
+    eot *= 229.18
+
+    # True local time
+    tlt = (day_time + longitude / 15 + eot / 60) % 24 - 12
```

Comment:
I suspect this is similar as [`hour_angle`](https://github.com/pvlib/pvlib-python/blob/b20ca5319fd2ea01b0261401d112b70f49146604/pvlib/solarposition.py#L1324) - perhaps there is a way to reuse the existing code, or modify it to suit your needs so we don't have duplicate code?

Reply:
looking in more detail they are the nearly the same:
    tlt = hour_angle / 15.

but IMO this version here, looks a little easier to read, with the use of `mod()`, so might be nice to replace the existing `hour_angle` with this, assuming there is better or same performance

Reply:
I tried `hour_angle` and it was not working in this code.

Reply:
But do you agree they are the same equation, that only differ by a factor of 1/15? we have noticed that different formulations have different "offsets" so perhaps adjusting the input may help? Or perhaps leave it,  raise an issue to collapse this later, and maybe add a `#TODO` in the code.  If @cwhanse and @wholmgren are okay then so am I